front: drop sec2time()

emersion opened the pull request:
secToHoursString() does exactly the same thing, but without manual string mungling and magic offsets.

Part of https://github.com/OpenRailAssociation/osrd/issues/10877

emersion committed:
front: drop sec2time()

secToHoursString() does exactly the same thing, but without
manual string mungling and magic offsets.

Signed-off-by: Simon Ser <contact@emersion.fr>

achrafmohye commented:
Indeed `secToHoursString` is safer :) but don't know if we need to better handle falsy values in `secToHoursString` i.e 0 so it returns `"00:00:00" `instead of empty string as it is the case in `sec2time`? and if it is a plausible scenario for a future usage as of now LGTM :)

SharglutDev reviewed (approved):
Lgtm and tested (although the input table is very buggy)

> [!NOTE]
> We should change the `secToHoursString` function docs

theocrsb reviewed (approved):
lgtm

emersion committed:
front: remove null special case in secToHoursString()

This function is never called with null. The zero handling was
incorrect for midnight departures/arrivals.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: fix outdated secToHoursString() docs

The withSeconds parameter is no more.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion commented:
> don't know if we need to better handle falsy values in secToHoursString i.e 0 so it returns "00:00:00" instead of empty string as it is the case in sec2time?

Good catch! I don't believe `secToHoursString()` should behave like this so I've updated it to stop doing that.

emersion commented:
> We should change the secToHoursString function docs

Done!
